homeassistant/components/soundtouch/media_player.py

```diff
@@ -234,6 +237,19 @@ def state(self):
 
         return MAP_STATUS.get(self._status.play_status, STATE_UNAVAILABLE)
 
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self._status.source
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return [
+            Source.AUX.value,
+            Source.BLUETOOTH.value,
```

Comment:
I presume Source is an enum? I'd recommend using something like:
```
return [x.value for x in list(Source)]
```
to avoid the need to keep this in sync when/if upstream changes and adds new potential sources.

Reply:
It is indeed, but it looks like most of them don't have a method for selecting it (which kind of links to your second comment)

https://github.com/CharlesBlonde/libsoundtouch/blob/25163a7ae39de2724e2946e8b1b61484d2a590d6/libsoundtouch/utils.py#L43-50